backend/app/utils/decorators.py

```diff
@@ -0,0 +1,27 @@
+"""Custom decorators for role-based access control."""
+from functools import wraps
+from flask_jwt_extended import verify_jwt_in_request, get_jwt_identity
+from flask import abort
+from ..models.user import User
+from ..extensions import DB as db
+
+def roles_required(*required_roles):
+    """Decorator to check if a user has one of the required roles."""
```

Comment:
Personally, when I see a code with conditionals or iterations,  I am like, there's something wrong here. And if it's with cascaded conditionals or iterations... that's a signal for a serious refactoring.